doc/RFC-MMP.llnl.04.txt

```diff
@@ -0,0 +1,913 @@
+=====================================================================
+2016-07-28 Draft 4		ZFS Multi-{Modifier,Mount} Protection
+=====================================================================
+Originally created 2009 by ricardo.correia@oracle.com
+  ** references 1
+Revised 2017 by {faaland1,bass6,behlendorf1}@llnl.gov
+=====================================================================
+
+As stated in RFE 6710718 (multi-import protection for the DMU), it would be
```

Comment:
I see some design docs / notes intermingled with the source code in illumos, e.g. `usr/src/uts/common/avs/ns/sdbc/cache_kstats_readme.txt`.  But I'm fine with creating a new directory in ZoL, it's up to @behlendorf .